### PR TITLE
Disable Automated Tests - LRAC-11657 | master

### PR DESCRIPTION
**Pull request by manuferreira**
https://issues.liferay.com/browse/LRAC-11657 (disable ticket)
https://issues.liferay.com/browse/LRAC-11651 (hide CSV)

### CODE REVIEW

**Comment by manuferreira** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase [hunk 1]
```diff
@@ -527,8 +527,12 @@ definition {
 	}
 
 	@description = "Feature ID: Legacy | Automation ID: LRAC-8594 | Test Summary: Deleting a data source causes individuals to be removed from segment"
+	@ignore = "true"
 	@priority = "4"
 	test DeleteDataSourceIndividualsRemovedSegment {
+
+		// CSV data source tests are no longer possible to run due to this ticket LRAC-11651
+
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase:535]: Hey @manuferreira, I think this test doesn't need to use CSV. I think it would be you have a segment with an individual and when you delete the data source, the individuals will be removed from the segment and the property. This test will still need to continue in quarantine as the membership list takes a long time to update.

I wanted @joshitagaki to confirm that this test could be done without CSV.

If he confirms, then I think you can undo this change and create a ticket for us to refactor this test or you can do it yourself in your ticket

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/IndividualsAttributeBreakdown.testcase [hunk 2]
```diff
@@ -144,8 +144,12 @@ definition {
 	}
 
 	@description = "Feature ID: Legacy | Automation ID: LRAC-8658 | Test Summary: Delete a new breakdown"
+	@ignore = "true"
 	@priority = "5"
 	test DeleteNewBreakdown {
+
+		// CSV data source tests are no longer possible to run due to this ticket LRAC-11651
+
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/IndividualsAttributeBreakdown.testcase:152]: Hey @manuferreira, This test can be done using another breakdown attribute that does not need the CSV (Ex: givenName, email and etc)

I think you can undo this change and create a ticket for us to refactor this test or you can do it yourself in your ticket

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase [hunk 3]
```diff
@@ -353,8 +353,12 @@ definition {
 	}
 
 	@description = "Feature ID: Legacy | Automation ID: LRAC-8481 | Test Summary: Create a Dynamic Individuals Segment with a criteria that uses a boolean and a does not contain"
+	@ignore = "true"
 	@priority = "4"
 	test CanCreateDynamicSegmentWithIndividualCriteriaUsesBooleanAndDoesNotContain {
+
+		// CSV data source tests are no longer possible to run due to this ticket LRAC-11651
+
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase:361]: Hey @manuferreira, I think this test can create a boolean type custom field to be used in place of the attribute that use the value from the CSV.

I think you can undo this change and create a ticket for us to refactor this test or you can do it yourself in your ticket

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase [hunk 4]
```diff
@@ -473,8 +477,12 @@ definition {
 	}
 
 	@description = "Feature ID: Legacy | Automation ID: LRAC-8480 | Test Summary: Create a Dynamic Individuals Segment with a criteria that uses is known and is not"
+	@ignore = "true"
 	@priority = "4"
 	test CanCreateDynamicSegmentWithIndividualCriteriaUsesIsKnownAndIsNot {
+
+		// CSV data source tests are no longer possible to run due to this ticket LRAC-11651
+
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase:485]: Hey @manuferreira, I think this validation can be done using other criteria.

I think you can undo this change and create a ticket for us to refactor this test or you can do it yourself in your ticket

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase [hunk 5]
```diff
@@ -944,8 +952,12 @@ definition {
 	}
 
 	@description = "Feature ID: Legacy | Automation ID: LRAC-8479 | Test Summary: Create a Segment with a Number Property"
+	@ignore = "true"
 	@priority = "4"
 	test CanCreateSegmentWithNumberProperty {
+
+		// CSV data source tests are no longer possible to run due to this ticket LRAC-11651
+
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase:960]: Hey @manuferreira, I think this validation can be done using other criteria.

I think you can undo this change and create a ticket for us to refactor this test or you can do it yourself in your ticket

Comment (ViniLopes20) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase:960]: This ticket is probable to have problems with automation because of https://issues.liferay.com/browse/LRAC-9329

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsDistribution.testcase [hunk 6]
```diff
@@ -145,6 +145,7 @@ definition {
 	@priority = "3"
 	test CanDistributionChartFilteredNumber {
 
+		// CSV data source tests are no longer possible to run due to this ticket LRAC-11651
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsDistribution.testcase:148]: Hey @manuferreira, I think this validation can be done using other criteria.

I think you can undo this change and create a ticket for us to refactor this test or you can do it yourself in your ticket

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsOverviewAttributeBreakdown.testcase [hunk 7]
```diff
@@ -201,6 +205,7 @@ definition {
 	@priority = "3"
 	test SegmentOverviewDistributionFilteredByNumber {
 
+		// CSV data source tests are no longer possible to run due to this ticket LRAC-11651
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsOverviewAttributeBreakdown.testcase:208]: Hey @manuferreira, I think this validation can be done using other criteria.

This ticket is probable to have problems with automation because of https://issues.liferay.com/browse/LRAC-9329

I think you can undo this change and create a ticket for us to refactor this test or you can do it yourself in your ticket

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsOverviewAttributeBreakdown.testcase [hunk 8]
```diff
@@ -36,8 +36,12 @@ definition {
 	}
 
 	@description = "Feature ID: Legacy | Automation ID: LRAC-8504 | Test Summary: Segment Overview distribution filtered by boolean"
+	@ignore = "true"
 	@priority = "3"
 	test SegmentOverviewDistributionFilteredByBoolean {
+
+		// CSV data source tests are no longer possible to run due to this ticket LRAC-11651
+
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsOverviewAttributeBreakdown.testcase:44]: Hey @manuferreira, I think this test can create a boolean type custom field to be used in place of the attribute that use the value from the CSV.

I think you can undo this change and create a ticket for us to refactor this test or you can do it yourself in your ticket

**Review by ViniLopes20 — COMMENTED** [review]
Hey @manuferreira, I think you forgot to ignore this test here:

`DataSourcesDeletionCSV#DeleteCSVDataSourceAndAssertDetailsFromDataSourceRemoved`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase (hunk 1 above)
Comment (joshitagaki) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase:535]: Correct. We don't need a CSV for this test.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase (hunk 5 above)
Comment (manuferreira) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase:960]: This bug that you pointed out is a static segment that is being created and later on a distribution filter and the test that I'm editing now `CanCreateSegmentWithNumberProperty` creates only a dynamic segment with an age field. So I don't think this test will have issues, or am I missing something? what do you think?

Comment (ViniLopes20) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase:960]: Ohh yes, I think this bug won't be a problem in this test. Maybe the delay of the membership list only

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsOverviewAttributeBreakdown.testcase (hunk 7 above)
Comment (christianstokes) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsOverviewAttributeBreakdown.testcase:208]: Thank you for highlighting LRAC-9329 @ViniLopes20. We will need to remember this issue until we can fully migrate to BQ.

I think it's okay to just disable these tests quickly to prevent failures, but as @ViniLopes20 said, a ticket should be created to refactor it.

Probably the easiest way to fix this would be to sync contacts with additional attributes if the required attribute type is not already in the default list of synced attributes.  I believe we can also create custom attributes to sync if we need any attribute types that are not in DXP by default.  Please message me if you have questions about this!

Comment (manuferreira) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsOverviewAttributeBreakdown.testcase:208]: I started to fix all these tests that need an update in this PR by creating a custom attribute on DXP (when it's necessary) this way we can maintain the idea of the test. 
Thanks, guys @christianstokes @ViniLopes20

**Comment by manuferreira** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase [hunk 9]
```diff
@@ -588,9 +580,9 @@ definition {
 		task ("Add a jobTitle field") {
 			ACSegments.goToSidebarAttributes(criterion = "Individual Attributes");
```
Comment (christianstokes) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase:581]: The "task" still says "Add a jobTitle field" even though it is using email

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase [hunk 10]
```diff
@@ -611,10 +603,8 @@ definition {
 			ACIndividualsDashboard.assertIndividualAttribute(individualAttribute = "New Segment Test");
 		}
 
-		task ("Delete the CSV Data Source") {
-			ACSettings.deleteDataSource(
-				dataSourceName = "CSV Job Title",
-				dataSourceType = "CSV");
```
Comment (christianstokes) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase:617]: This is a complicated test, because it involves checking that segment membership updates when only *some* of an individual's attributes are removed. Specifically those attributes which gave them segment membership in the first place.

How we test this now:
1. Combine individual attributes from different data sources into one individual, like so:
| Data Source 1 | Individual 1 | Attribute A |
| Data Source 2 | Individual 1 | Attribute B |
2. Create a segment with criteria based on only one attribute (Attribute B)
3. Delete Data Source 2 where the individual got Attribute B from
4. Check to see if Individual 1 is no longer in the segment, because their Attribute B was deleted

So if I understand the change to the test, it follows these steps:

1. Add a data source with a DXP user
2. Add a segment with the criteria "email = ac@liferay.com"
3. Go to the individual and assert that segments tab shows that they belong to the segment you created
4. Delete the data source that the individual is from
5. Go to individual's segment tab and assert that they are no longer associated with the segment.

Does this actually work? I would expect that when you do step 4, it would delete the user completely, meaning we could no longer check their membership?
Or maybe I am missing something.

**Review by christianstokes — COMMENTED** [review]
Awesome fixes @manuferreira. Great use of custom fields to achieve the same thing as CSV. I have some questions about the first test which you can see in the comment, but the others look good to me.

I understand we still have problems with the membership list being too slow so some of these tests might fail anyways, is that right?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase (hunk 9 above)
Comment (manuferreira) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase:581]: Ohh nice, thanks for noticing that!
I was reviewing here and I'm seeing I forgot to add some minor things. I will fix it soon.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase (hunk 10 above)
Comment (ViniLopes20) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase:617]: Yes, you're right. The individual will be excluded from the property. This verification after the deletion is necessary to navigate inside the created segment and verify that in the membership list there is no data

Comment (manuferreira) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase:617]: You are right Christian, thank you. I'm sending the changes :smile:

**Comment by manuferreira** [comment]
> Awesome fixes @manuferreira. Great use of custom fields to achieve the same thing as CSV. I have some questions about the first test which you can see in the comment, but the others look good to me.
> 
> I understand we still have problems with the membership list being too slow so some of these tests might fail anyways, is that right?

Yes, you are right.

**Comment by manuferreira** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase [hunk 11]
```diff
@@ -611,28 +603,20 @@ definition {
...
+		task ("Open the created segment and assert the user is not present") {
+			ACNavigation.goToSegments();
 
-			ACNavigation.switchTab(tabName = "Segments");
+			ACSegments.accessSegment(segmentName = "New Segment Test");
 
-			AssertElementNotPresent(
-				key_individualAttribute = "New Segment Test",
-				locator1 = "ACIndividualsDashboard#KNOWN_INDIVIDUAL_ATTRIBUTE");
+			ACUtils.viewNameListIsNotPresent(nameList = "ac ac");
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase:619]: Hey @manuferreira, this macro will not work if the result of the list is an empty state message. This path waits for a format with classes that only appear when there is a list on the page, if you had another user appearing in the list this macro would work, but as it will display the empty state message, so I think you could use:

`ACUtils.viewGenericText(textValueList = "There are no members found on the selected time period.");`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase [hunk 12]
```diff
@@ -567,16 +569,6 @@ definition {
 			ACProperties.switchProperty(propertyName = "${propertyName}");
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase:569]: Hey @manuferreira, I think you need to use the `assignedPropertyName` variable:

`ACProperties.switchProperty(propertyName = "${assignedPropertyName}");`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase [hunk 13]
```diff
@@ -1043,7 +1077,7 @@ definition {
 			ACSegments.viewSegmentCriteriaIndividualAttributes(
 				attributeName = "age",
 				attributeValue = "40",
-				condition = "less than");
+				condition = "greater than");
 		}
 
 		task ("Switch tab to membership and check the membership list") {
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase:1083]: Hey @manuferreira, I think at the end of the test you could put it to verify that user2 is not in the list, that way we can verify that the criteria are working in the segment:

`ACUtils.viewNameListIsNotPresent(nameList = "user2 user2");`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase [hunk 14]
```diff
@@ -574,21 +592,21 @@ definition {
...
 				condition = "is known",
 				index = "2");
 		}
 
 		task ("Switch tab to membership and check the membership list") {
 			ACNavigation.switchTab(tabName = "Membership");
 
-			ACUtils.viewNameList(nameList = "${membershipList}");
+			ACUtils.viewNameList(nameList = "user3 user3");
 		}
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase:610]: Hey @manuferreira, I think at the end of the test you could put it to verify that user1 and user2 are not in the list, that way we can verify that the criteria are working in the segment:

`for (var username : list "user1 user1,user2 user2") {
         ACUtils.viewNameListIsNotPresent(nameList = "${username}");
}`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase [hunk 15]
```diff
@@ -453,13 +469,13 @@ definition {
...
+				attributeValue = '''"testing@liferay.com"''',
 				condition = "does not contain",
 				conjunction = "AND");
 
 			ACSegments.viewSegmentCriteriaIndividualAttributes(
-				attributeName = "doNotCall",
+				attributeName = "User Custom Field",
 				attributeValue = "FALSE",
 				condition = "is",
 				index = "2");
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase:481]: Hey @manuferreira, I think at the end of the test you could put it to verify that user1 and user2 are not in the list, that way we can verify that the criteria are working in the segment:

`for (var username : list "user1 user1,user2 user2") {
         ACUtils.viewNameListIsNotPresent(nameList = "${username}");
}`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsOverviewAttributeBreakdown.testcase [hunk 16]
```diff
@@ -105,14 +109,14 @@ definition {
...
-				attribute = "doNotCall",
+				attribute = "User Custom Field",
 				breakdownName = "Segment Overview Test");
 		}
 
 		task ("View Breakdown Chart") {
 			ACIndividualsDashboard.viewBreakdownRecharts(
 				fillColor = "#4B9BFF",
-				tick = "false");
+				tick = "true");
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsOverviewAttributeBreakdown.testcase:119]: Hey @manuferreira, I think you missed putting this value as TRUE in the user account configuration in DXP or is this not necessary?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase (hunk 12 above)
Comment (manuferreira) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DeleteSegments.testcase:569]: Oh yeah, I forgot to update this line.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase (hunk 15 above)
Comment (manuferreira) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase:481]: I think inside the ACUtils.viewNameListIsNotPresent we already have a for loop.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase (hunk 14 above)
Comment (manuferreira) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase:610]: I think inside the ACUtils.viewNameListIsNotPresent we already have a for loop.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase (hunk 15 above)
Comment (ViniLopes20) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreation.testcase:481]: Ohh yes, thanks Manu

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsOverviewAttributeBreakdown.testcase (hunk 16 above)
Comment (manuferreira) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsOverviewAttributeBreakdown.testcase:119]: Oh you mean, when I'm adding the custom attribute for the user I should set it to True?
The default value for the custom attribute that I created is True, so no need to set it again in the user configuration.

Comment (ViniLopes20) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsOverviewAttributeBreakdown.testcase:119]: Ohhh ok, thanks for the explanation, Manu 😄

**Comment by manuferreira** [comment]
ci:test:sf

**Review by ViniLopes20 — APPROVED** [review]
Amazing job updating and ignoring all these cases. Very good LGTM 🎉

**Review by ViniLopes20 — APPROVED** [review]

**Review by christianstokes — APPROVED** [review]
Lots of changes, very nice! Thank you for working on all of these tests @manuferreira

**Comment by christianstokes** [comment]
ci:forward
